src/circles.rs

```diff
@@ -128,25 +138,38 @@ fn get_colored_circles (center_to_center: i32, small_radius: i32, colors: Vec<[u
     out
 }
 
-/// function to calculate contents of the png image with 19-circle icon
-pub fn calculate_png_data (big_radius: i32, colors: Vec<[u8; 4]>) -> Vec<u8> {
-    
+/// Calculate png image pixel data (only pixel colors)
+///
+/// Requires image size in pixels (equal to diameter of largest, outer circle),
+/// and identicon colors
+#[cfg(feature = "pix")]
+pub fn calculate_png_data(size_in_pixels: u16, colors: [[u8; 4]; 19]) -> Vec<u8> {
     let mut data: Vec<u8> = Vec::new();
-    let small_radius = big_radius/32*5;
-    let center_to_center = big_radius/8*3;
-    
+    let big_radius = size_in_pixels as f32 / 2.0;
+    let small_radius = big_radius / 32.0 * 5.0;
+    let center_to_center = big_radius / 8.0 * 3.0;
+
     let big_circle = Circle {
-        x_center: 0,
-        y_center: 0,
+        x_center: 0.0,
+        y_center: 0.0,
         radius: big_radius,
         rgba_color: FOREGROUND_COLOR,
     };
-    
-    let small_circles_set = get_colored_circles (center_to_center, small_radius, colors);
-    
-    for y in -big_radius..big_radius+1 {
-        for x in -big_radius..big_radius+1 {
-            if in_circle (x, y, &big_circle) {
+
+    let small_circles_set = get_colored_circles(center_to_center, small_radius, colors);
+
+    let iter_start = -(size_in_pixels as i32) / 2;
+    let iter_end = {
+        if size_in_pixels % 2 == 0 {
+            size_in_pixels as i32 / 2
+        } else {
+            size_in_pixels as i32 / 2 + 1
+        }
+    };
+
+    for y in iter_start..iter_end {
```

Comment:
Adding a comment that this draws the circle pixel by pixel would be appreciated.